### PR TITLE
Using jib instead of docker-maven-plugin 

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

**Comment by andxu** [comment]
Given the jib solution is more straight forward and easy for containerizing the spring-boot project and  building the image, I suggest we switch to jib for building/pushing image.

**Comment by openpublishbuild** [comment]
OPS Build status updates of commit _[951ba1b](https://github.com/andxu/azure-docs-sdk-java/pull/788/commits/951ba1be10b454d39a90a9a338c172b23a039cb5)_: 

:clock10: **Full build**: the average full build time is _69 min(s) 43 sec(s)_, this is based on the last _155_ build(s) for this repository.

 :warning: **This is a FULL build** due to the incremental build being disabled in configuration file (.openpublishing.publish.config.json in 'live' branch). For more detailed info, please click [here](https://aka.ms/full-build-reasons).

**Comment by openpublishbuild** [comment]
OPS Build status updates of commit _[951ba1b](https://github.com/andxu/azure-docs-sdk-java/pull/788/commits/951ba1be10b454d39a90a9a338c172b23a039cb5)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs-ref-conceptual/spring-framework/deploy-spring-boot-java-app-on-kubernetes.md](https://github.com/andxu/azure-docs-sdk-java/blob/andy_jib/docs-ref-conceptual/spring-framework/deploy-spring-boot-java-app-on-kubernetes.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/java/azure/spring-framework/deploy-spring-boot-java-app-on-kubernetes?view=azure-java-stable&branch=pr-en-us-788) (azure-java-stable)<br/> [View](https://review.docs.microsoft.com/en-us/java/azure/spring-framework/deploy-spring-boot-java-app-on-kubernetes?view=azure-java-preview&branch=pr-en-us-788) (azure-java-preview)<br/> [View](https://review.docs.microsoft.com/en-us/java/azure/spring-framework/deploy-spring-boot-java-app-on-kubernetes?view=azure-java-legacy&branch=pr-en-us-788) (azure-java-legacy) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C3%5C18%5C5de83821-f44b-0e93-473e-c19393f102fe%5CPullRequest%5C201903180440549368-788%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=yPFBygX%2By9%2F96Q5L2aQjC8cciKP2i2OQCXxUhiVnfOg%3D&st=2019-03-18T05%3A51%3A22Z&se=2019-04-18T05%3A56%3A22Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by openpublishbuild** [comment]
OPS Build status updates of commit _[1fb89cd](https://github.com/andxu/azure-docs-sdk-java/pull/788/commits/1fb89cdf7fccb0015e8458a2d293bbcbcae23866)_: 

:clock10: **Full build**: the average full build time is _69 min(s) 44 sec(s)_, this is based on the last _156_ build(s) for this repository.

 :warning: **This is a FULL build** due to the incremental build being disabled in configuration file (.openpublishing.publish.config.json in 'live' branch). For more detailed info, please click [here](https://aka.ms/full-build-reasons).

**Comment by openpublishbuild** [comment]
OPS Build status updates of commit _[1fb89cd](https://github.com/andxu/azure-docs-sdk-java/pull/788/commits/1fb89cdf7fccb0015e8458a2d293bbcbcae23866)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs-ref-conceptual/spring-framework/deploy-spring-boot-java-app-on-kubernetes.md](https://github.com/andxu/azure-docs-sdk-java/blob/andy_jib/docs-ref-conceptual/spring-framework/deploy-spring-boot-java-app-on-kubernetes.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/java/azure/spring-framework/deploy-spring-boot-java-app-on-kubernetes?view=azure-java-stable&branch=pr-en-us-788) (azure-java-stable)<br/> [View](https://review.docs.microsoft.com/en-us/java/azure/spring-framework/deploy-spring-boot-java-app-on-kubernetes?view=azure-java-preview&branch=pr-en-us-788) (azure-java-preview)<br/> [View](https://review.docs.microsoft.com/en-us/java/azure/spring-framework/deploy-spring-boot-java-app-on-kubernetes?view=azure-java-legacy&branch=pr-en-us-788) (azure-java-legacy) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C3%5C18%5C5de83821-f44b-0e93-473e-c19393f102fe%5CPullRequest%5C201903180441431865-788%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=FFe8h2cK54hYLtpObqTmwXhClySVewG4DIOLwGK%2B%2BXw%3D&st=2019-03-18T07%3A01%3A58Z&se=2019-04-18T07%3A06%3A58Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs-ref-conceptual/spring-framework/deploy-spring-boot-java-app-on-kubernetes.md [hunk 1]
```diff
@@ -200,8 +156,34 @@ The `id` and `username` are the name of the registry. Use the `password` value f
    ```
    This command may take a while to complete.
 
-1. When you're using Azure Container Registry (ACR) with Azure Kubernetes Service (AKS), an authentication mechanism needs to be established. Follow the steps in [Authenticate with Azure Container Registry from Azure Kubernetes Service] to grant AKS access to ACR.
-
+1. When you're using Azure Container Registry (ACR) with Azure Kubernetes Service (AKS), you need to grant Azure Kubernetes Service pull access to Azure Container Registry. Azure creates a service principal when you are creating Azure Kubernetes Service to support cluster operability with other Azure resources. You can use this auto-generated service principal for authentication with an ACR registry, using the following scripts in PowerShell to grant the AKS-generated service principal pull access to an Azure container registry, please refer to
```
Comment (testforstephen) [inline docs-ref-conceptual/spring-framework/deploy-spring-boot-java-app-on-kubernetes.md]: You can use this auto-generated service principal for authentication with an ACR registry, using the following scripts in PowerShell to grant the AKS-generated service principal pull access to an Azure container registry, please ...
->
Please run the following scripts in PowerShell to grant the AKS-generated service principal pull access to an Azure Container Registry, see more details [Authenticate with Azure Container Registry from Azure Kubernetes Service].

**Comment by openpublishbuild** [comment]
OPS Build status updates of commit _[eaefc7a](https://github.com/andxu/azure-docs-sdk-java/pull/788/commits/eaefc7a10b372a29de9cdcfc357a9193df955bf8)_: 

:clock10: **Full build**: the average full build time is _69 min(s) 43 sec(s)_, this is based on the last _159_ build(s) for this repository.

 :warning: **This is a FULL build** due to the incremental build being disabled in configuration file (.openpublishing.publish.config.json in 'live' branch). For more detailed info, please click [here](https://aka.ms/full-build-reasons).

**Comment by openpublishbuild** [comment]
OPS Build status updates of commit _[eaefc7a](https://github.com/andxu/azure-docs-sdk-java/pull/788/commits/eaefc7a10b372a29de9cdcfc357a9193df955bf8)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs-ref-conceptual/spring-framework/deploy-spring-boot-java-app-on-kubernetes.md](https://github.com/andxu/azure-docs-sdk-java/blob/andy_jib/docs-ref-conceptual/spring-framework/deploy-spring-boot-java-app-on-kubernetes.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/java/azure/spring-framework/deploy-spring-boot-java-app-on-kubernetes?view=azure-java-stable&branch=pr-en-us-788) (azure-java-stable)<br/> [View](https://review.docs.microsoft.com/en-us/java/azure/spring-framework/deploy-spring-boot-java-app-on-kubernetes?view=azure-java-preview&branch=pr-en-us-788) (azure-java-preview)<br/> [View](https://review.docs.microsoft.com/en-us/java/azure/spring-framework/deploy-spring-boot-java-app-on-kubernetes?view=azure-java-legacy&branch=pr-en-us-788) (azure-java-legacy) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C3%5C19%5C5de83821-f44b-0e93-473e-c19393f102fe%5CPullRequest%5C201903190443212311-788%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=FEGfl4KPvawT6ROreLs4FpOtmUr7unM4DDAJXuW%2F%2Bs0%3D&st=2019-03-19T05%3A51%3A59Z&se=2019-04-19T05%3A56%3A59Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by openpublishbuild** [comment]
OPS Build status updates of commit _[ba302cc](https://github.com/andxu/azure-docs-sdk-java/pull/788/commits/ba302cc6e8b2b20e9704ec5c46f83cddbad4bfd2)_: 

:clock10: **Full build**: the average full build time is _69 min(s) 42 sec(s)_, this is based on the last _160_ build(s) for this repository.

 :warning: **This is a FULL build** due to the incremental build being disabled in configuration file (.openpublishing.publish.config.json in 'live' branch). For more detailed info, please click [here](https://aka.ms/full-build-reasons).

**Comment by openpublishbuild** [comment]
OPS Build status updates of commit _[ba302cc](https://github.com/andxu/azure-docs-sdk-java/pull/788/commits/ba302cc6e8b2b20e9704ec5c46f83cddbad4bfd2)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs-ref-conceptual/spring-framework/deploy-spring-boot-java-app-on-kubernetes.md](https://github.com/andxu/azure-docs-sdk-java/blob/andy_jib/docs-ref-conceptual/spring-framework/deploy-spring-boot-java-app-on-kubernetes.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/java/azure/spring-framework/deploy-spring-boot-java-app-on-kubernetes?view=azure-java-stable&branch=pr-en-us-788) (azure-java-stable)<br/> [View](https://review.docs.microsoft.com/en-us/java/azure/spring-framework/deploy-spring-boot-java-app-on-kubernetes?view=azure-java-preview&branch=pr-en-us-788) (azure-java-preview)<br/> [View](https://review.docs.microsoft.com/en-us/java/azure/spring-framework/deploy-spring-boot-java-app-on-kubernetes?view=azure-java-legacy&branch=pr-en-us-788) (azure-java-legacy) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C3%5C19%5C5de83821-f44b-0e93-473e-c19393f102fe%5CPullRequest%5C201903190455029650-788%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=qUZI1ZPU%2F2BqU1fnarXj0NmGmm%2F5I8FiRAP%2BQeRPcpQ%3D&st=2019-03-19T07%3A06%3A55Z&se=2019-04-19T07%3A11%3A55Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by kyliel** [comment]
@andxu , could you please help refactor the whole tutorial? Thank you.

1. Do we really need this note? This tutorial requires your environment can install docker. It does not request you open virtualization feature locally. Suggest to remove this note. 

 Note
Due to the virtualization requirements of this tutorial, you cannot follow the steps in this article on a virtual machine; you must use a physical computer with virtualization features enabled.

2. Change below title to "Create a Spring Boot application". That section shows you how to git clone a gs-spring-boot-docker, build it and run it locally. It does not show how to containerize it. Through this sample project is spring-boot-docker, no need to highlight the sample applciation name in the title. 

Title: Create the Spring Boot on Docker Getting Started web app

3. Remove below step since it is a common step. No need to teach developers how to create and enter a directory in our tutorial. 

Open a command-prompt and create a local directory to hold your application, and change to that directory; for example:


Copy
md C:\SpringBoot
cd C:\SpringBoot
-- or --


Copy
md /users/robert/SpringBoot
cd /users/robert/SpringBoot

4. Navigate to the completed project directory for your Spring Boot application (for example, "C:\SpringBoot\gs-spring-boot-docker\complete" or "/users/robert/SpringBoot/gs-spring-boot-docker/complete"), and open the pom.xml file with a text editor.

-> /users/xxx/SpringBoot/gs-spring-boot-docker/complete or C:\SpringBoot\gs-spring-boot-docker\complete

5. server name of your Azure Container Registry -> registry name of your Azure Container Registry

6. Please run the following scripts in PowerShell to grant the AKS-generated service principal pull access to an Azure Container Registry, see more details at Authenticate with Azure Container Registry from Azure Kubernetes Service.

-> Please run the following scripts in bash or Powershell to grant AKS access  to ACR, see more details at ...

7. No need to have below line since it shows "PowerShell' in the box of scripts.

 # Please run the following scripts in PowerShell.

8. Move "bash" ahead of "PowerShell". 

9. The numbers of the chapter "Create a Kubernetes Cluster on AKS using the Azure CLI" are 1, 2 and then 1, 2. Please fix the sequence.

10. Move "deploy with kubectl" ahead of 'Deploy with Kubernetes web interface" since "This tutorial deploys the app using kubectl, then allow you to explore the deployment through the Kubernetes web interface."

11. Remove below link in Next steps/Additional Resources. Below link is talking about App Service Container support instead of Azure Container Service. 
Deploy a Spring Boot application on Linux in the Azure Container Service

12. Add one line in additional resources.
For more information about iteratively running and debugging containers directly in Azure Kubernetes Service (AKS) with Azure Dev Spaces, see [get started on Azure Dev Spaces with Java](https://docs.microsoft.com/azure/dev-spaces/get-started-java)

**Comment by openpublishbuild** [comment]
OPS Build status updates of commit _[9760b13](https://github.com/andxu/azure-docs-sdk-java/pull/788/commits/9760b13a6d7be4e4cf6bdfadb87346d4572f8c32)_: 

:clock10: **Full build**: the average full build time is _69 min(s) 40 sec(s)_, this is based on the last _174_ build(s) for this repository.

 :warning: **This is a FULL build** due to the incremental build being disabled in configuration file (.openpublishing.publish.config.json in 'live' branch). For more detailed info, please click [here](https://aka.ms/full-build-reasons).

**Comment by openpublishbuild** [comment]
OPS Build status updates of commit _[9760b13](https://github.com/andxu/azure-docs-sdk-java/pull/788/commits/9760b13a6d7be4e4cf6bdfadb87346d4572f8c32)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs-ref-conceptual/spring-framework/deploy-spring-boot-java-app-on-kubernetes.md](https://github.com/andxu/azure-docs-sdk-java/blob/andy_jib/docs-ref-conceptual/spring-framework/deploy-spring-boot-java-app-on-kubernetes.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/java/azure/spring-framework/deploy-spring-boot-java-app-on-kubernetes?view=azure-java-stable&branch=pr-en-us-788) (azure-java-stable)<br/> [View](https://review.docs.microsoft.com/en-us/java/azure/spring-framework/deploy-spring-boot-java-app-on-kubernetes?view=azure-java-preview&branch=pr-en-us-788) (azure-java-preview)<br/> [View](https://review.docs.microsoft.com/en-us/java/azure/spring-framework/deploy-spring-boot-java-app-on-kubernetes?view=azure-java-legacy&branch=pr-en-us-788) (azure-java-legacy) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C3%5C28%5C5de83821-f44b-0e93-473e-c19393f102fe%5CPullRequest%5C201903280457576287-788%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=PjLdCo48qizOQO1ubV46u3bcC2CuqgSXaI%2FWfzGHPus%3D&st=2019-03-28T06%3A07%3A40Z&se=2019-04-28T06%3A12%3A40Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by andxu** [comment]
signoff

**Comment by andxu** [comment]
#sign-off

**Pull request merged**